### ISSUE TITLE
Minidriver crash if p15card->app is not set

### ISSUE THREAD

**Issue by jozsefd** [OpenSC/OpenSC#3084]
### Problem Description

If the card is using a PKCS15 Emulator where the optional pointer to the sc_app_info in the sc_pkcs15_card structure is not set then the minidriver crashes. The function get_pin_by_name will crash, the log will look like:

  P:10156; T:9336 2024-03-20 15:48:59.567 [cardmod] MD_Function:md_get_pin_by_role:541 called
  P:10156; T:9336 2024-03-20 15:48:59.567 [cardmod] MD_Function:get_pin_by_name:499 called

### Proposed Resolution

Add a check for the pointer. All other references to this pointer check it already.

```C
	if ( p15card->app != NULL ) {
		sc_bin_to_hex(p15card->app->path.value, p15card->app->path.len, str_path, sizeof(str_path), 0);
		blocks = scconf_find_blocks(p15card->card->ctx->conf, conf_block, "application", str_path);
	}
```

### Steps to reproduce

Insert a card which uses an emulator, not setting the p15card->app (e.g. nqApplet) and try to load the minidriver, like "certutil /scinfo".

### Logs

  P:10156; T:9336 2024-03-20 15:48:59.567 [cardmod] MD_Function:md_get_pin_by_role:541 called
  P:10156; T:9336 2024-03-20 15:48:59.567 [cardmod] MD_Function:get_pin_by_name:499 called

**Comment by Jakuje** [comment]
Can you propose the change as a PR?

**Comment by jozsefd** [comment]
For sure, I have already created a branch with the fixes of #3084 #3085 and #3086. I will double check it as #3077 gets pulled.

**Comment by Jakuje** [comment]
I think this part is not fixed in #3077 as it is not completely related. Can you open a separate PR for this fix?

**Comment by jozsefd** [comment]
Opened PR #3095 with the suggested fix.
